prov/efa/src/efa_msg.c

```diff
@@ -309,6 +309,12 @@ static ssize_t efa_post_send(struct efa_ep *ep, const struct fi_msg *msg, uint64
 
 	efa_post_send_sgl(ep, msg, ewr);
 
+	if (flags & FI_INJECT)
+		if (OFI_LIKELY(len <= ep->domain->ctx->inline_buf_size))
```

Comment:
We can possibly avoid the extra if here as ibv_post_send will validate the inline size as well.
Is there added value in exiting early here?

Reply:
Good point, there is no real value in returning early. I'll drop the validation.

Reply:
Done.